docs/File-Formats.md

```diff
@@ -1310,3 +1311,63 @@ The cells contain treatment response values that can be postive and negative rea
+
+### Generic Assay meta file
+The generic assay metadata file should contain the following fields:
+```
+cancer_study_identifier: Same value as specified in meta file of the study
+genetic_alteration_type: GENERIC_ASSAY
+generic_assay_type: <GENERIC_ASSAY_TYPE>, e.g., "TREATMENT" or "MUTATIONAL_SIGNATURE"
+datatype: LIMIT-VALUE
+stable_id: Any unique identifier using a combination of alphanumeric characters, _ and -
+profile_name: A name describing the analysis.
+profile_description: A description of the data processing done.
+data_filename: <name of generic assay data file>
+show_profile_in_analysis_tab: true
+generic_entity_meta_properties: A comma separate list of generic entity properties, e.g., "name,description"
+```
+
+Example:
+```
+cancer_study_identifier: study_es_0
+genetic_alteration_type: GENERIC_ASSAY
+generic_assay_type: MUTATIONAL_SIGNATURE
+datatype: LIMIT-VALUE
+stable_id: mutational_signature
+profile_name: data of mutational signature
+profile_description: Description of data of mutational signature
+data_filename: data_mutational_signature.txt
+show_profile_in_analysis_tab: true
+generic_entity_meta_properties: name,description
+```
+
+#### Note on `stable_id`
+For generic assay, meta files values for `stable_id` are not pre-registered in the cBioPortal implementation, but can be freely chosen by the user. Requirement is that each generic assay meta file is assigned a unique `stable_id` by the user. This requirement is validated by the data validation scripts.
+
+#### Note on `generic_entity_meta_properties`
+All meta properties must be specified in `generic_entity_meta_properties`. Also, all meta properties specified in `generic_entity_meta_properties` are required to be shown in the data file as same name columns. These fields are free text (e.g., 'name of genetic entity').
+
+#### Note on `Generic Assay` genetic_alteration_type and `LIMIT-VALUE` datatype
+All generic assay data is registered to be of the type of `genetic_alteration_type` and data type `LIMIT-VALUE`. This alteration and data type is intended to be used for any numerical data set with similar structure (entities measured in samples). The `LIMIT-VALUE` is validated to contain any continuous number optionally prefixed with a '>' or '<' threshold symbol (e.g., '>8.00'). If the value for the generic entity in the respective sample could not (or was not) be measured (or detected), the value should be 'NA'.
+
+### Generic Assay data file
+The data file will be a simple tab separated format, similar to the expression data file: each sample is a column, each generic entity is a row, each cell contains values for that generic entity x sample combination.
+
+For each generic entity (row) in the data file, the following columns are required in the order specified:
+
+- ***entity_stable_id***: Any unique identifier using a combination of alphanumeric characters, _ and -.
```

Comment:
Please uppercase all the column names from here on - entity_stable_id, name, description.